client/db/bolt/db.go

```diff
@@ -437,12 +438,33 @@ func (db *BoltDB) UpdateOrder(m *dexdb.MetaOrder) error {
 	if len(md.Proof.DEXSig) == 0 {
 		return fmt.Errorf("cannot save order without DEX signature")
 	}
-	return db.ordersUpdate(func(master *bbolt.Bucket) error {
+	return db.ordersUpdate(func(ob, aob *bbolt.Bucket) error {
 		oid := ord.ID()
-		oBkt, err := master.CreateBucketIfNotExists(oid[:])
-		if err != nil {
-			return fmt.Errorf("order bucket error: %w", err)
+
+		// Create or move an order bucket based on order status. Orders
+		// that have status less than or equal to booked go in the
+		// ordersBucket. Finished orders go in the archivedOrdersBucket.
+		var oBkt *bbolt.Bucket
+		var err error
+		executed := md.Status > order.OrderStatusBooked
+		if executed {
+			if ob.Bucket(oid[:]) != nil {
+				ob.DeleteBucket(oid[:])
+			}
+			oBkt, err = aob.CreateBucketIfNotExists(oid[:])
+			if err != nil {
+				return fmt.Errorf("order bucket error: %w", err)
+			}
+		} else {
+			if aob.Bucket(oid[:]) != nil {
+				return fmt.Errorf("archived order %s set to active", oid)
+			}
+			oBkt, err = ob.CreateBucketIfNotExists(oid[:])
+			if err != nil {
+				return fmt.Errorf("order bucket error: %w", err)
```

Comment:
Let's supplement these `CreateBucketIfNotExists` error strings with "archived" and "active"